misc/python/materialize/ci_util/upload_debug_symbols_to_polarsignals.py

```diff
@@ -0,0 +1,247 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+import os
+import subprocess
+import tempfile
+from pathlib import Path
+from tempfile import _TemporaryFileWrapper
+
+import boto3
+
+from materialize import mzbuild, spawn, ui
+from materialize.ci_util.upload_debug_symbols_to_s3 import (
+    DEBUGINFO_BINS,
+    DEBUGINFO_S3_BUCKET,
+)
+from materialize.mzbuild import Repository, ResolvedImage
+from materialize.rustc_flags import Sanitizer
+
+
+def main() -> None:
+    coverage = ui.env_is_truthy("CI_COVERAGE_ENABLED")
+    sanitizer = Sanitizer[os.getenv("CI_SANITIZER", "none")]
+    bazel = ui.env_is_truthy("CI_BAZEL_BUILD")
+    bazel_remote_cache = os.getenv("CI_BAZEL_REMOTE_CACHE")
+
+    repo = mzbuild.Repository(
+        Path("."),
+        coverage=coverage,
+        sanitizer=sanitizer,
+        bazel=bazel,
+        bazel_remote_cache=bazel_remote_cache,
+    )
+
+    collect_and_upload_debug_data_to_polarsignals(repo, DEBUGINFO_BINS)
+
+
+def collect_and_upload_debug_data_to_polarsignals(
+    repo: mzbuild.Repository, debuginfo_bins: set[str]
+) -> None:
+    is_tag_build = ui.env_is_truthy("BUILDKITE_TAG")
+
+    # Upload debuginfo and sources to Polar Signals (our continuous
+    # profiling provider), but only if this is a tag build. Polar Signals is
+    # expensive, so we don't want to upload development or unstable builds
+    # that won't ever be profiled by Polar Signals.
+
+    if not is_tag_build:
+        print("Uploading no data to PolarSignals because we are not on a tag build")
+        return
```

Comment:
Skipping here works, but I think what we probably want is something in `ci/mkpipeline.py` that trims this step away

Reply:
It's more a second line of defense that is actually not necessary.

We have this filter in the template file:
```
        if: build.tag != "" && build.branch =~ /^v.*\..*\$/
```

I think I should remove the check in the Python code since it is not fully in line with the condition in the template file.

Reply:
✅